.teamcity/test/AllProjectsTest.kt

```diff
@@ -31,18 +31,47 @@ class AllProjectsTest {
                 ProjectRelease to "ProjectKtorRelease"
         )
 
-        projectsFromPackage("subprojects").forEach { project ->
+        allProjects().forEach { project ->
             assertTrue(projectsIDs.containsKey(project), "Cannot find expected ID for project ${project.name}")
             assertEquals(projectsIDs[project], project.id.toString(), "ID for Project ${project.name} doesn't match")
             assertHasUniqueID(project)
         }
     }
 
+    @Test
+    fun projectsBuildTypesHadDefinedIDs() {
+        val buildTypesIDs = mapOf(
+                ProjectCore to setOf(
+                        "KtorMatrixMacOSJava8",
+                        "KtorMatrixMacOSJava11",
+                        "KtorMatrixLinuxJava8",
+                        "KtorMatrixLinuxJava11",
+                        "KtorMatrixWindowsJava8",
+                        "KtorMatrixWindowsJava11"
+                ),
+                ProjectDocSamples to setOf("KtorDocs_ValidateSamples")
+        )
+
+        allProjects().forEach { project ->
+            if (project.buildTypes.size > 0) {
+                assertTrue(buildTypesIDs.containsKey(project), "Cannot find build types IDs for project ${project.name}")
+                val ids = project.buildTypes.map { it.id.toString() }.toSet()
+                assertEquals(buildTypesIDs[project], ids, "Build types IDs for project ${project.name} doesn't match")
+            }
+        }
+    }
+
     private fun assertHasUniqueID(project: Project) {
         val className = project.javaClass.name.split('.').last()
         assertNotEquals(className, project.id.toString(), "Project ID '${project.id.toString()}' should differ from its class name")
```

Comment:
While this is similar to the assertion below, this could be classified as an assertion helper, given the prefix, but is the line that extracts classname failsafe?

Reply:
Could you clarify this is similar to what assertion?
The line that extracts class name could fail.

Reply:
Sorry, I mean the "allProjects" below (in this code review) that was acting as an assertion.

Reply:
I've moved assertion inside `allProjects` to the separate test